c/driver/common/utils.c

```diff
@@ -124,19 +124,27 @@ void StringBuilderInit(struct StringBuilder* builder, size_t initial_size) {
   builder->size = 0;
   builder->capacity = initial_size;
 }
-void StringBuilderAppend(struct StringBuilder* builder, const char* value) {
-  size_t length = strlen(value);
-  size_t new_size = builder->size + length;
-  if (new_size > builder->capacity) {
-    size_t new_capacity = builder->size + length - builder->capacity;
-    if (builder->size == 0) new_capacity++;
-
-    builder->buffer = realloc(builder->buffer, new_capacity);
-    builder->capacity = new_capacity;
+void StringBuilderAppend(struct StringBuilder* builder, const char* fmt, ...) {
+  va_list argptr;
+  int bytes_available = builder->capacity - builder->size;
+
+  va_start(argptr, fmt);
+  int n = vsnprintf(builder->buffer + builder->size, bytes_available, fmt, argptr);
+  va_end(argptr);
+
+  if (n < 0) {                        // TODO: handle error
```

Comment:
Should we handle the TODO here?

Reply:
Yea we should. Ideally these aren't void functions but return an integer to indicate success / failure. 

Without trying to do too much at once I am thinking of that as a follow up, but can try to tackle within this too if it doesn't make the diff too large

Reply:
A follow up is fine (just make sure to file it so it can be `TODO(...)`